src/users/UserManager.ts

```diff
@@ -38,7 +38,16 @@ export default class UserManager {
 
     return response.data;
   }
-
+  static async getUserAsStreamerInfoById(userId: string): Promise<StreamerInfo> {
+    return this.getUserById(userId).then((twitchUser) => {
```

Comment:
If you're using async then use await, otherwise there's no point using the async keyword. Returning a promise without the async keyword will work the same as with it in this case.

Reply:
My lack of async/await/Promise knowledge shines through... So the async keyword will automatically wrap the function in a Promise? And it will be resolved by the caller?
I just wanted to keep the existing behaviour of not calling the Twitch API until actually needed.

Reply:
async/await is the new way of allowing you to use the resolved value inline, rather than chaining/callback heck. It essentially tells the ECMA engine to wrap up the promises, and let you use them more in-line rather than needing all the .then's, which makes code easier to follow, but under the hood it's still promises. If you don't use await, then adding async doesn't do much, or anything. The receiving function can still await if it's an async function because backwards compatibility with old-school promises is important.